contributors/design-proposals/autoscaling/hpa-external-metrics.md

```diff
@@ -0,0 +1,144 @@
+
+External and Object metrics will specify the desired target by setting exactly one of two fields: TargetValue (global target) or TargetAverageValue (per-pod target).
+
+
+# Multiple metric values
+
+Label selector specified by user can match multiple time series, resulting in multiple values provided to HPA. In such case the sum of all those values will be used for autoscaling. This is meant to allow autoscaling using a metric that is drilled down by some criteria not relevant for autoscaling a particular workload (for example to allow autoscaling based on a total number of HTTP requests, regardless of which HTTP method is used).
+
+If the need arises we can easily add other simple aggregations and allow user to choose one of them.
+
+# Example
+
+This is an example HPA configuration autoscaling based on number of pending messages in a message queue (RabbtMQ) running outside of cluster.
+
+```yaml
+kind: HorizontalPodAutoscaler
+apiVersion: autoscaling/v2beta2
+spec:
+  scaleTargetRef:
+    kind: ReplicationController
+    name: Worker
+  minReplicas: 2
+  maxReplicas: 10
+  metrics:
+   - type: External
+     metricName: queue_messages_ready
+     metricSelector:
+       queue: worker_tasks
+     targetAverageValue: 30
+```
+
+# API
+
+This is the part of autoscaling/v2beta2 API that includes changes from v2beta1.
+
+Some parts containing obvious changes (MetricSpec and MetricStatus) have been omitted for clarity.
+
+```go
+// MetricSourceType indicates the type of metric.
+type MetricSourceType string
+
+var (
+        // ObjectMetricSourceType is a metric describing a Kubernetes object
+        // (for example, hits-per-second on an Ingress object).
+        ObjectMetricSourceType MetricSourceType = "Object"
+        // PodsMetricSourceType is a metric describing each pod in the current scale
+        // target (for example, transactions-processed-per-second).  The values
+        // will be averaged together before being compared to the target value.
+        PodsMetricSourceType MetricSourceType = "Pods"
+        // ResourceMetricSourceType is a resource metric known to Kubernetes, as
+        // specified in requests and limits, describing each pod in the current
+        // scale target (e.g. CPU or memory).  Such metrics are built in to
+        // Kubernetes, and have special scaling options on top of those available
+        // to normal per-pod metrics (the "pods" source).
+        ResourceMetricSourceType MetricSourceType = "Resource"
+        // ExternalMetricSourceType is a global metric coming from
+        // outside of Kubernetes. It allows autoscaling based on information
+        // coming from components running outside of cluster
```

Comment:
this is a bit of a documentation issue, but I think "coming from outside the cluster" is the wrong term here.  I think it's better to stress that this is for the case when the metrics aren't immediately associatable with objects inside the cluster.

For instance, I could run a DBaaS that runs outside the cluster, but still provide metrics to track usage per namespace, and thus not have to use the `external` type.

Reply:
Good point. Updated all comments as per your suggestion.